module/bdev/raid/bdev_raid.c

```diff
@@ -2196,6 +2271,384 @@ _raid_bdev_remove_base_bdev(struct raid_base_bdev_info *base_info,
+
+	if (ctx->cb_fn != NULL) {
+		ctx->cb_fn(ctx->cb_arg, ctx->status);
+	}
+
+	free(ctx);
+}
+
+static void
+raid_bdev_channel_stop_faulty_state_done(struct spdk_io_channel_iter *i, int status)
+{
+	struct raid_bdev_delta_bitmap_ctx *ctx = spdk_io_channel_iter_get_ctx(i);
+	struct raid_bdev *raid_bdev = ctx->base_info->raid_bdev;
+
+	ctx->status = status;
+	spdk_bdev_unquiesce(&raid_bdev->bdev, &g_raid_if,
+			    raid_bdev_stop_faulty_state_done, ctx);
+}
+
+static void
+raid_bdev_channel_stop_faulty_state(struct spdk_io_channel_iter *i)
+{
+	struct spdk_io_channel *ch = spdk_io_channel_iter_get_channel(i);
+	struct raid_bdev_io_channel *raid_ch = spdk_io_channel_get_ctx(ch);
+	struct raid_bdev_delta_bitmap_ctx *ctx = spdk_io_channel_iter_get_ctx(i);
+	struct raid_base_bdev_info *base_info = ctx->base_info;
+	int rc = 0;
+
+	rc = base_info->raid_bdev->module->channel_faulty_base_bdev(base_info, raid_ch,
+			BASE_BDEV_STATE_FAULTY_STOPPED);
+
+	spdk_for_each_channel_continue(i, rc);
+}
+
+static void
+_raid_bdev_base_bdev_stop_faulty_state(void *_ctx, int status)
+{
+	struct raid_bdev_delta_bitmap_ctx *ctx = _ctx;
+	struct raid_base_bdev_info *base_info = ctx->base_info;
+	struct raid_bdev *raid_bdev = base_info->raid_bdev;
+
+	if (status != 0) {
+		SPDK_ERRLOG("Failed to quiesce raid bdev %s: %s\n",
+			    raid_bdev->bdev.name, spdk_strerror(-status));
+		raid_bdev_stop_faulty_state_done(ctx, status);
+		return;
+	}
+
+	spdk_for_each_channel(raid_bdev, raid_bdev_channel_stop_faulty_state, ctx,
+			      raid_bdev_channel_stop_faulty_state_done);
+}
+
+/*
+ * brief:
+ * raid_bdev_stop_base_bdev_delta_bitmap function is the method to stop the updating of the delta
+ * bitmap for a faulty base bdev.
+ * If the base bdev is not part of any raid, or if the base bdev is not in faulty state, false is
+ * returned.
+ * params:
+ * base_bdev_name - name of the base bdev
+ * returns:
+ * true - success
+ * false - failure
+ */
+int
+raid_bdev_stop_base_bdev_delta_bitmap(char *base_bdev_name, raid_bdev_destruct_cb cb_fn,
+				      void *cb_arg)
+{
+	struct raid_bdev_delta_bitmap_ctx *ctx;
+	struct raid_base_bdev_info *base_info;
+	struct raid_bdev *raid_bdev;
+	struct spdk_bdev *bdev;
+
+	assert(spdk_get_thread() == spdk_thread_get_app_thread());
+
+	base_info = raid_bdev_find_faulty_base_info_by_name(base_bdev_name);
+
+	if (!base_info) {
+		SPDK_ERRLOG("bdev '%s' not a faulty base info of any RAID\n", base_bdev_name);
+		return -ENODEV;
+	}
+
+	if (base_info->state == BASE_BDEV_STATE_FAULTY_STOPPED) {
+		SPDK_ERRLOG("Delta bitmap of base bdev %s already stopped", base_bdev_name);
+		return -EEXIST;
+	}
+
+	ctx = calloc(1, sizeof(*ctx));
+	if (ctx == NULL) {
+		SPDK_ERRLOG("Unable to allocate memory for delta bitmap stop context of base bdev %s\n",
+			    base_bdev_name);
+		return -ENOMEM;
+	}
+
+	raid_bdev = base_info->raid_bdev;
+	bdev = &raid_bdev->bdev;
+	base_info->delta_bitmap = spdk_bit_array_create(bdev->blockcnt / bdev->optimal_io_boundary);
```

Comment:
`bdev->blockcnt / bdev->optimal_io_boundary` is commonly used in the codes. Can we add a field, e.g. regioncnt,  for it?

Reply:
The region's block number is a concept used only in the raid, so I have created a function in this module.